src/customers/customers.controller.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Post,
+  Put,
+} from '@nestjs/common';
+import { CustomersService } from './customers.service';
+import { CreateCustomerDto } from '../dto/create-customer.dto';
+import { UpdateCustomerDto } from '../dto/update-customer.dto';
+
+@Controller('customers')
+export class CustomersController {
+  constructor(private readonly customersService: CustomersService) {}
+
+  @Post()
+  create(@Body() createCustomerDto: CreateCustomerDto) {
+    return this.customersService.create(createCustomerDto);
+  }
+
+  @Get()
+  index() {
+    return this.customersService.getAllCustomers();
+  }
+
+  @Get(':id')
+  show(@Param('id') id: string) {
+    return this.customersService.getCustomerById(+id);
+  }
+
+  @Put(':id')
+  update(
+    @Param('id') id: string,
+    @Body() updateCustomerDto: UpdateCustomerDto,
+  ) {
+    return this.customersService.updateCustomerById(+id, updateCustomerDto);
+  }
```

Comment:
This should be PATCH and not a PUT as we allow the consumer of the API for a partial update of `Customer` entity (unlike the `Appointment` entity where you need to provide all properties to update the object)